buildpipeline/Dotnet-CoreClr-Trusted-BuildTests.json

```diff
@@ -85,80 +85,28 @@
-        "nugetConfigPath": "",
-        "externalEndpoints": "",
-        "noCache": "false",
-        "packagesDirectory": "",
-        "verbosityRestore": "Detailed",
-        "searchPatternPush": "$(Build.ArtifactStagingDirectory)/**/*.nupkg;!$(Build.ArtifactStagingDirectory)/**/*.symbols.nupkg",
-        "nuGetFeedType": "internal",
-        "feedPublish": "",
-        "allowPackageConflicts": "false",
-        "externalEndpoint": "",
-        "verbosityPush": "Detailed",
-        "searchPatternPack": "**/*.csproj",
-        "configurationToPack": "$(BuildConfiguration)",
-        "outputDir": "$(Build.ArtifactStagingDirectory)",
-        "versioningScheme": "off",
-        "includeReferencedProjects": "false",
-        "versionEnvVar": "",
-        "requestedMajorVersion": "1",
-        "requestedMinorVersion": "0",
-        "requestedPatchVersion": "0",
-        "packTimezone": "utc",
-        "includeSymbols": "false",
-        "toolPackage": "false",
-        "buildProperties": "",
-        "verbosityPack": "Detailed",
-        "arguments": "install Microsoft.DotNet.Build.Tasks.ScepOps -source https://devdiv.pkgs.visualstudio.com/_packaging/dotnet-core-internal-tooling/nuget/v3/index.json -OutputDirectory $(Build.StagingDirectory)"
-      }
-    },
     {
       "environment": {},
       "enabled": true,
       "continueOnError": true,
       "alwaysRun": false,
-      "displayName": "Uninstall SCEP",
+      "displayName": "Uninstall SCEP if script exists",
       "timeoutInMinutes": 0,
-      "refName": "Task52",
+      "condition": "succeededOrFailed()",
       "task": {
-        "id": "c6c4c611-aa2e-4a33-b606-5eaba2196824",
-        "versionSpec": "1.*",
+        "id": "e213ff0f-5d5c-4791-802d-52ea3e7be1f1",
+        "versionSpec": "2.*",
         "definitionType": "task"
       },
       "inputs": {
-        "solution": "$(Build.StagingDirectory)\\Microsoft.DotNet.Build.Tasks.ScepOps.1.0.1\\build\\Microsoft.DotNet.Build.Tasks.ScepOps.targets",
-        "msbuildLocationMethod": "version",
-        "msbuildVersion": "14.0",
-        "msbuildArchitecture": "x64",
-        "msbuildLocation": "",
-        "platform": "",
-        "configuration": "",
-        "msbuildArguments": "/t:UninstallSCEP",
-        "clean": "false",
-        "maximumCpuCount": "false",
-        "restoreNugetPackages": "false",
-        "logProjectEvents": "false",
-        "createLogFile": "false"
+        "targetType": "inline",
+        "filePath": "",
+        "arguments": "-Action u -ToolsPath  $(ScepToolsPath)",
+        "script": "if (Test-Path \"$(ScepToolsPath)SCEP_Tools\\scripts\\Microsoft.DotNet.Build.Tasks.Extensions.ps1\") {\n     \"Microsoft.DotNet.Build.Tasks.Extensions.ps1 script found. Executing action...\"\n    & $(ScepToolsPath)SCEP_Tools\\scripts\\Microsoft.DotNet.Build.Tasks.Extensions.ps1 -Action u -ToolsPath $(ScepToolsPath)\n} else {\n   \"Microsoft.DotNet.Build.Tasks.Extensions.ps1 script does not exist. Moving on...\"\n}",
```

Comment:
I'm slightly confused by the name of the file `Microsoft.DotNet.Build.Tasks.Extensions.ps`1.  It's so generic that it makes me think it's meant as an entry point for multiple types of extensions.  At the moment, it appears that there is only one extension **and** that this file lives under the `SCEP_Tools` folder which implies that this file will only ever apply to `SCEP_Tools`.  What's the intention behind this layout?  If the ps1 file is general purpose, then it should probably be at some higher level.

Reply:
The idea is that it can hold different sort of "extension" tasks not just SCEP, but I get your point and we can change the structure if and when we add more things to it.

Reply:
IMO it would be less work to change the structure now rather than have a bunch of repos dependent on this structure and trying to refactor all the build defs which are going to need to change.  Even if, initially, it was something like...
```
\DotNetCore
  - Microsoft.DotNet.Buildt.Tasks.Extensions.ps1
  \SCEP_Tools
    \bin
    - SCEP_Tools.ps1
```

It would be nice if we could say for certain that every extension was either a `start/stop` or `install/uninstall` scenario.  Probably the former, and then the root script can iterate through it's known extensions (maybe based on some token string for which ones to iterate through) and either start or stop them based on the verb passed to the entry script.

That's just off the top of my head, @weshaggard might have additional thoughts / experience here.

Reply:
I like this being a directory that has a general structure:

```
AgentTools\
           \Begin.ps1
           \End.ps1
```
These scripts will be the wrappers the definitions depend on and can be edited to call whatever tasks we need such as SCEP tool and agent clean-up tasks, etc. 

If you guys don't like Begin/End we could do Pre/Post. 

We should use a similar pattern for non-windows but instead have a Begin.sh and End.sh script.

Reply:
Or we could follow MSBuild convention and go with Before/After.  Begin/End sounds reasonable to me too though.

Reply:
Before/After is fine with me as well.